### PR TITLE
Fixes to Section 3 ("verdi shell").

### PR DESCRIPTION
**Pull request by borellim**
Some questions:
* Please confirm that the tutorial will take place under Python 3. If not, or if we are not sure, we should change the instructions for creating a new Jupyter notebook to reflect this.
* We say that `it is recommended (and easier) to use the DataFactory function instead of an explicit import`. To be honest I was never convinced of this, especially now that we limit importable depths in aiida-core to 2 levels: direct imports offer auto-completion in many editors, and this is a huge benefit (esp. for newbies). Can you try to make the argument more convincing?
* I explained that `Dict` was once called `ParameterData`: hope it makes sense.

Thanks!

### CODE REVIEW

**Comment by sphuber** [comment]
As per your questions:
* Yes the current VM is running python 3
* The reason was and still is (despite, the new module hierarchy import simplification) that it only relies on the plugin name, which should be unchanging. The modules however aren't and even though currently the inner module structure could change and through the exposing to the top, one could still use `aiida.orm` directly, at some point some of the data classes may actually be removed from `aiida-core`, such as `StructureData`. If one uses the `DataFactory` now, the code will continue to work without changes
* Was there still a reference to `ParameterData` anywhere? If not, would this perhaps create more confusion than it clears up?

File: docs/pages/2019_MARVEL_Psik_MaX/sections/verdi_shell.rst [hunk 1]
```diff
@@ -113,7 +113,7 @@ k-points
 ~~~~~~~~
 
 A set of k-points in the Brillouin zone is represented by an instance of the
-``KpointsData`` class. Choose one from the graph of Section :ref:`aiidagraph`,
+``KpointsData`` class. Choose one from the graph of :numref:`aiidagraph`,
```
Comment (sphuber) [inline docs/pages/2019_MARVEL_Psik_MaX/sections/verdi_shell.rst]: The link `aiidagraph` is actually a section link, not the figure. If you want to reference the figure use `fig_graph`

File: docs/pages/2019_MARVEL_Psik_MaX/sections/verdi_shell.rst [hunk 2]
```diff
@@ -163,8 +163,9 @@ the kpoints class itself!
 Parameters
 ~~~~~~~~~~
 
-Dictionaries with various parameters are represented in AiiDA by ``Dict`` nodes.
-Get the PK and load the input parameters of a calculation in the graph of Section :ref:`aiidagraph`.
+Dictionaries with various parameters are represented in AiiDA by ``Dict`` nodes
+(formerly called ``ParameterData`` in previous versions of AiiDA).
+Get the PK and load the input parameters of a calculation in the graph of :numref:`aiidagraph`.
```
Comment (sphuber) [inline docs/pages/2019_MARVEL_Psik_MaX/sections/verdi_shell.rst]: wrong link, see above

File: docs/pages/2019_MARVEL_Psik_MaX/sections/verdi_shell.rst [hunk 3]
```diff
@@ -201,7 +202,7 @@ the verdi shell, you will be prompted with a string notifying you the
 Structures
 ~~~~~~~~~~
 
-Find a structure in the graph of Section :ref:`aiidagraph` and load it.
+Find a structure in the graph of :numref:`aiidagraph` and load it.
```
Comment (sphuber) [inline docs/pages/2019_MARVEL_Psik_MaX/sections/verdi_shell.rst]: idem

File: docs/pages/2019_MARVEL_Psik_MaX/sections/verdi_shell.rst [hunk 4]
```diff
@@ -231,14 +232,14 @@ with lattice parameter `a`\ :sub:`lat`\ `= 5.4` Å:
      Default units for crystal structure cell and coordinates in AiiDA are Å
      (Ångström).
 
-Structures in AiiDA are instances of ``StructureData`` class: load it in the
+Structures in AiiDA are instances of class ``StructureData``: load it in the
```
Comment (sphuber) [inline docs/pages/2019_MARVEL_Psik_MaX/sections/verdi_shell.rst]: instance of *the* class

**Review by sphuber — CHANGES_REQUESTED** [review]
Looking good, just a few minor changes

**Comment by borellim** [comment]
Thanks a lot for replying.

> at some point some of the data classes may actually be removed from `aiida-core`

Makes sense. So it's "more future-proof" rather than "safer"... no?

> Was there still a reference to ParameterData anywhere? If not, would this perhaps create more confusion than it clears up?

Possibly, but if I was a student and I already knew AiiDA from pre-1.0, I would want to see this change highlighted.
Anyway, I found some references, but I don't know if these pages still need to be rewritten:
```
~/repos/aiida-tutorials/docs/pages/2019_MARVEL_Psik_MaX$ grep -rin ParameterData
sections/appendix_input_validation.rst:49:    parameters = ParameterData(dict=validated_dict)
sections/verdi_shell.rst:167:(formerly called ``ParameterData`` in previous versions of AiiDA).
scripts/equation_of_states.py:50:        ParameterData = DataFactory("parameter")
scripts/equation_of_states.py:53:                'result': ParameterData(dict={'eos_data': eos})
scripts/pressure_convergence.py:16:ParameterData = DataFactory('parameter')
notebooks/querybuilder-template.ipynb:382:    " ParameterData | 2922  \n",
```

**Comment by sphuber** [comment]
> Makes sense. So it's "more future-proof" rather than "safer"... no?

Yes, that is a better way of putting it. Feel free to adapt the text to say that

> Possibly, but if I was a student and I already knew AiiDA from pre-1.0, I would want to see this change highlighted.

Fair enough, but then shouldn't we also do the same for the many other changes?
There are way too many of those and it would clutter the text.

> Anyway, I found some references, but I don't know if these pages still need to be rewritten:
> 
> ```
> ~/repos/aiida-tutorials/docs/pages/2019_MARVEL_Psik_MaX$ grep -rin ParameterData
> sections/appendix_input_validation.rst:49:    parameters = ParameterData(dict=validated_dict)
> sections/verdi_shell.rst:167:(formerly called ``ParameterData`` in previous versions of AiiDA).
> scripts/equation_of_states.py:50:        ParameterData = DataFactory("parameter")
> scripts/equation_of_states.py:53:                'result': ParameterData(dict={'eos_data': eos})
> scripts/pressure_convergence.py:16:ParameterData = DataFactory('parameter')
> notebooks/querybuilder-template.ipynb:382:    " ParameterData | 2922  \n",
> ```

Those are problems and need to be fixed.

**Comment by borellim** [comment]
> > Makes sense. So it's "more future-proof" rather than "safer"... no?
> 
> Yes, that is a better way of putting it. Feel free to adapt the text to say that

Done. Please take a look if you like it.

> > Possibly, but if I was a student and I already knew AiiDA from pre-1.0, I would want to see this change highlighted.
> 
> Fair enough, but then shouldn't we also do the same for the many other changes?
> There are way too many of those and it would clutter the text.

Ok, fine. You know better than me how many of these cases there are. I have removed the reference to `ParameterData`.

**Comment by ltalirz** [comment]
@sphuber I let you do the final review & merge

**Comment by sphuber** [comment]
Thanks to both

**Pull request merged**